### DISCUSSION TITLE
Possible to Link Local Higher Quality Assets?

### DISCUSSION THREAD

**Discussion by CameraTraveler27** [mozilla/hubs#3414] [General Questions & Discussion] [answered]
Hi I'm trying to play higher quality, longer form 360 videos as well as bring in higher resolution 3D models within Hubs shared user online spaces. 

I know that Hubs has a very low cap to these file sizes but was hoping its possible to get around that by having each user in the room predownload these much larger assets to each other their local computers before entering the room, link those assets to Hubs and then use Hubs for mostly it's avatars and positional tracking.

The end goal is create a best of both worlds scenario where multiple users are still able to see and talk to each other's avatars in Hubs over common wifi speeds while still being able to have a shared experience of much higher quality assets because they exist on each of their local machines. 

Is this possible?

**Comment by netpro2k** [comment, marked as answer]
It's not possible to playback local content, but the upload limit can be worked around by simply hosting the video files somewhere else publicly accessible on the internet. This could be your own web server, or something like Github, google drive, Dropbox, Amazon S3, etc. The browser should handle buffering the video and downloading parts as it plays it back, so there should be no need to pre download the video. With your own web server you can also configure the cache settings so that the users browsers will correctly download and store a cached copy of the video for some time so that they wont need to re-download it if they refresh the page.

For example: https://file-examples-com.github.io/uploads/2017/04/file_example_MP4_1920_18MG.mp4. You should be able to paste that url into Hubs or into the url field of a Spoke media element.
